pkg/clusternetwork/config.go

```diff
@@ -77,3 +127,14 @@ func getNetworkType(oclient configclient.Interface) (string, error) {
 	}
 	return networkCR.Spec.NetworkType, nil
 }
+
+// getServiceNetworkCIDR gets the serviceCIDR using cluster config
+// this is required for cni configuration
+func getServiceNetworkCIDR(oclient configclient.Interface) (string, error) {
+	// Get the cluster network object so that we can find the service network
+	networkCR, err := oclient.ConfigV1().Networks().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "error getting cluster network object")
+	}
```

Comment:
Should we validate  networkCR.Spec.ServiceNetwork ?

Reply:
I am validating if the value is a empty string as a part of GetServiceCIDR() method which is a part of clusterNetworkConfig interface and called in main.go. Are you suggesting, I add an empty string check here as well?